watertap/core/custom_spacer_base.py

```diff
@@ -0,0 +1,718 @@
+            doc="""This argument changes the correlation used to calculate the axial pressure drop on the feed side.
+    **default** - PressureCorrelation.guillen.
+    **Valid values:** {
+    no_spacer: Hagen-Poiseuille correlation is used for pressure drop correlation.
+    guillen: Guillen & Hoek correlation is used for pressure drop calculation. Standard case in WaterTAP model
+    schock: Schock & Miquel correlation is used for pressure drop calculation.
+    da_costa: DaCosta correlation is used for pressure drop calculation.
+    parameterized: Parameterizes the Power number 
+    References: https://doi.org/10.1016/j.cej.2008.10.030 - Guillen & Hoek
+                Da Costa, Andre. Fluid flow and mass transfer in spacer-filled
+                channels for ultrafiltration. Diss. UNSW Sydney, 1993. - DaCosta and no_spacer correlations
+                1) Equation 4.12 for no spacer pressure drop correlation
+                2) Equation 4.27 for DaCosta pressure drop correlation
+                    Assumptions:   Cd value is assumed as 10 based on the justification from DaCosta thesis page 116
+                    k_theta values in the third term is assumed to be 0.98 according to the assumption in DaCosta
+                    thesis for turbulent regime, the values for laminar flow are higher.
+                    NF is normalized to length so the overall dp_dx equation gives pressure drop per unit length}""",
+        ),
+    )
+
+
+class SpacerType(Enum):
+    """
+    no_spacer: Converts RO unit model to a case with no spacer
+    net_spacer: Adds necessary correlations for net_spacers
+    printed_spacer: Adds 3D printed spacer and necessary correlations
+    """
+
+    no_spacer = auto()
+    net_spacer = auto()
+    printed_spacer = auto()
+
+
+class SherwoodCorrelation(Enum):
+    """
+    graetz: Graetz-Leveque correlation is used for mass transfer correlation. Exclusive for no spacer case
+    sieder: Sieder-Tate correlation is used for mass transfer correlation. Exclusive for no spacer case
+    guillen: Guillen & Hoek correlation is used for mass transfer correlation. Standard case in WaterTAP model
+    schock: Schock & Miquel correlation is used for mass transfer correlation.
+    da_costa: DaCosta correlation is used for mass transfer correlation.
+    parameterized: Parameterizes the Sherwood number
+    References: https://doi.org/10.1016/j.cej.2008.10.030 - Guillen & Hoek
+                Da Costa, Andre. Fluid flow and mass transfer in spacer-filled
+                channels for ultrafiltration. Diss. UNSW Sydney, 1993. - DaCosta and no_spacer correlations
+                1) Table 5.1 for Graetz-Leveque and Sieder-Tate correlations
+                2) Equation 5.20 for DaCosta Sherwood correlation
+    """
+
+    graetz = auto()
+    sieder = auto()
+    guillen = auto()
+    schock = auto()
+    da_costa = auto()
+    parameterized = auto()
```

Comment:
Given the breadth of additions in this PR, we'll definitely want to add documentation to readthedocs to capture all the additions. That can come at the end of this though, once satisfied with the core code changes.